src/PathSimulator.cpp

```diff
@@ -0,0 +1,368 @@
+    bool        rightSingle      = false;
+    std::size_t startElemBracket = bracketSize + 1;
+    std::size_t strayElem        = 0;
+    std::size_t memoryLeft       = 0;
+    std::size_t bracketMemory    = 0;
+    std::size_t opMemory         = 0;
+    //std::cout << warmupDepth << std::endl;
+    for (std::size_t i = 0; i < bracketSize; i++) {
+        if (i == 0)
+            path.emplace_back(0, 1);
+        else {
+            //std::cout << qc->getNops() + i << " " << 1+i << std::endl;
+            path.emplace_back(qc->getNops() + i, 1 + i);
+        }
+    }
+    memoryLeft = qc->getNops() + bracketSize;
+    while (!rightSingle) {
+        for (auto i = 0U; i < bracketSize - 1; i++) {
+            if (startElemBracket == qc->getNops()) {
+                rightSingle = true;
+                strayElem   = startElemBracket;
+                break;
+            }
+            if (i == 0) {
+                //std::cout << startElemBracket << " " << startElemBracket+1 << std::endl;
+                path.emplace_back(startElemBracket, startElemBracket + 1);
+                opMemory++;
+                if (startElemBracket + 1 == qc->getNops()) {
+                    strayElem   = qc->getNops() + bracketSize + 1 + (bracketSize * bracketMemory - bracketMemory);
+                    rightSingle = true;
+                    break;
+                }
+            } else {
+                //  std::cout << qc->getNops() + warmupDepth+i+(bracketSize*bracketMemory-bracketMemory) << " " << startElemBracket+1+i << std::endl;
+                path.emplace_back(qc->getNops() + bracketSize + i + (bracketSize * bracketMemory - bracketMemory), startElemBracket + 1 + i);
+                opMemory++;
+                if (startElemBracket + 1 + i >= qc->getNops()) {
+                    strayElem   = qc->getNops() + bracketSize + i + (bracketSize * bracketMemory - bracketMemory) + 1;
+                    rightSingle = true;
+                    break;
+                }
+            }
+        }
+        if (rightSingle) {
+            break;
+        }
+        opMemory = 0;
+        startElemBracket += bracketSize;
+        bracketMemory++;
+    }
+    for (auto i = 0U; i < bracketMemory; i++) {
+        if (i == 0) {
+            //std::cout << memoryLeft << " " << memoryLeft+(bracketSize-1) << std::endl;
+            path.emplace_back(memoryLeft, memoryLeft + (bracketSize - 1));
+        } else {
+            //std::cout << memoryLeft + (bracketSize-1)*bracketMemory+opMemory +i << " " << memoryLeft+(bracketSize-1)*(i+1) << std::endl;
+            path.emplace_back(memoryLeft + (bracketSize - 1) * bracketMemory + opMemory + i, memoryLeft + (bracketSize - 1) * (i + 1));
+        }
+    }
+    if (rightSingle) {
+        //std::cout << memoryLeft + (bracketSize)*bracketMemory+opMemory << " " << strayElem << std::endl;
+        path.emplace_back(memoryLeft + (bracketSize)*bracketMemory + opMemory, strayElem);
+    }
+    setSimulationPath(path, true);
+}
```

Comment:
These functions could use some documentation comments so that maybe we understand what they are actually doing in a couple of months